### PR TITLE
Fix NotSerializableException when triggering BigDiffy through luigi

### PR DESCRIPTION
**Pull request by anne-decusatis**
exception before submitting to dataflow:

```
 Caused by: java.io.NotSerializableException: com.spotify.ratatool.diffy.BigDiffy$
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
INFO:   at java.io.ObjectOutputStream.writeArray(ObjectOutputStream.java:1378)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1174)
INFO:   at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
INFO:   at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
INFO:   at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
INFO:   at java.io.ObjectOutputStream.writeArray(ObjectOutputStream.java:1378)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1174)
INFO:   at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
INFO:   at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
INFO:   at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
INFO:   at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
INFO:   at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
INFO:   at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
INFO:   at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
INFO:   at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
INFO:   at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
INFO:   at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
INFO:   at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
INFO:   at org.apache.beam.sdk.util.SerializableUtils.serializeToByteArray(SerializableUtils.java:51)
INFO:   ... 28 more

```

I  confirmed this happens on ratatool 0.3.15 and successfully submits to dataflow on a local publish with these changes. I don't know how to reproduce outside luigi - it seems to work fine on the old version with sbt runMain with the same params.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -53,7 +53,8 @@ val commonSettings = Sonatype.sonatypeSettings ++ releaseSettings ++ Seq(
   managedSourceDirectories in Compile := (managedSourceDirectories in Compile).value
     .filterNot(_.getPath.endsWith("/src_managed/main")),
   javacOptions ++= Seq("-source", "1.8", "-target", "1.8", "-Xlint:unchecked"),
-  addCompilerPlugin(scalafixSemanticdb)
+  addCompilerPlugin(scalafixSemanticdb),
+  run / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.Flat
```
Comment (anne-decusatis) [inline build.sbt:57]: necessary to runMain and submit to dataflow

File: ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala [hunk 2]
```diff
@@ -130,8 +130,9 @@ case class FieldStats(field: String,
 }
 
 /** Big diff between two data sets given a primary key. */
-class BigDiffy[T : Coder](lhs: SCollection[T], rhs: SCollection[T],
-                          diffy: Diffy[T], keyFn: T => MultiKey, ignoreNan: Boolean = false) {
+class BigDiffy[T : Coder](@transient val lhs: SCollection[T], @transient val rhs: SCollection[T],
+                          diffy: Diffy[T], keyFn: T => MultiKey, ignoreNan: Boolean = false)
+  extends Serializable {
```
Comment (idreeskhan) [inline ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:135]: Does it work without this line but with the change on line 177?

File: build.sbt (hunk 1 above)
Comment (idreeskhan) [reply build.sbt:57]: Is this new?

Comment (anne-decusatis) [reply build.sbt:57]: new that it's needed for dataflow runner on sbt 1.3.x see https://stackoverflow.com/a/60341215

File: ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala (hunk 2 above)
Comment (anne-decusatis) [reply ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:135]: I tried removing `extends Serializable` from both and rerunning; it failed.

Comment (rscarvalho) [reply ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:135]: I believe only `case` classes are serializable by default (and have registered coders built-in in scio), hence the need to explicitly mark this class as `Serializable`. Great job find that out @anne-decusatis !

**Pull request merged**